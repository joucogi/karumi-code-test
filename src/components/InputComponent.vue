<template>
  <b-form-input :id="id" :type="type" :placeholder="placeholder"></b-form-input>
</template>
<script>
export default {
  props: {
    id: {
      type: String,
      default: "",
      required: false
    },
    type: {
      type: String,
      default: "text",
      required: false
    },
    placeholder: {
      type: String,
      default: "",
      required: false
    }
  }
};
</script>
<style scoped></style>
