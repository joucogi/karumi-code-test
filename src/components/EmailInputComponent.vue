<template>
  <InputComponent :id="id" type="email" :placeholder="placeholder" />
</template>
<script>
import InputComponent from "./InputComponent";
export default {
  components: { InputComponent },
  props: {
    id: {
      type: String,
      default: "",
      required: false
    },
    placeholder: {
      type: String,
      default: "",
      required: false
    }
  }
};
</script>
