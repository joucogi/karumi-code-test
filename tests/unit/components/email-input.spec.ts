import { shallowMount, Wrapper } from "@vue/test-utils";
import Vue from "vue";

import EmailInputComponent from "@/components/EmailInputComponent.vue";
import InputComponent from "@/components/InputComponent.vue";

import { checkInputProps, componentExists, toMatchSnapshot } from "../helpers";

let wrapper: Wrapper<Vue>;

beforeEach(() => {
  wrapper = shallowMount(EmailInputComponent);
});

describe("EmailInputComponent", () => {
  it("contents an InputComponent", () => {
    componentExists(wrapper, InputComponent);
  });

  it("shows an InputComponent with empty placeholder", () => {
    const input = wrapper.findComponent(InputComponent);
    checkInputProps(input, "", "email");
  });

  it("shows an InputComponent with Username text in placeholder", async () => {
    const placeholder = "Username";

    wrapper.setProps({
      placeholder
    });
    await wrapper.vm.$nextTick();

    const input = wrapper.findComponent(InputComponent);
    checkInputProps(input, placeholder, "email");
  });

  it("renders correctly", () => {
    toMatchSnapshot(wrapper);
  });
});
