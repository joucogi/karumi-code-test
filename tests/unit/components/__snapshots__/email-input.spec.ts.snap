// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInputComponent renders correctly 1`] = `
<inputcomponent-stub
  id=""
  placeholder=""
  type="email"
/>
`;
